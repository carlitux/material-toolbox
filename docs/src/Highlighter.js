// @flow
/* eslint-disable */
import * as React from 'react';
import Highlight  from 'react-syntax-highlighter';
import { monokaiSublime as monokai } from 'react-syntax-highlighter/dist/styles/hljs';
/* eslint-enable */

type Props = {
  children: string,
  language: string,
};

const Highlighter = ({ children, language }: Props) => (
  <Highlight showLineNumbers language={language} style={monokai}>
    {children}
  </Highlight>
);

Highlighter.defaultProps = {
  language: 'bash', // eslint-disable-line
};

export default Highlighter;
