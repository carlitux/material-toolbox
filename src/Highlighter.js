// @flow
import * as React from 'react';
import Highlight from 'react-syntax-highlighter'; // eslint-disable-line
import { monokaiSublime as monokai } from 'react-syntax-highlighter/styles/hljs'; // eslint-disable-line

type Props = {
  children: string,
  language?: string,
};

const Highlighter = ({ children, language }: Props) => (
  <Highlight showLineNumbers language={language} style={monokai}>
    {children}
  </Highlight>
);

Highlighter.defaultProps = {
  language: 'bash',
};

export default Highlighter;
