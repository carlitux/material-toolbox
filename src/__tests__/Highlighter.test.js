import React from 'react';
import { shallow } from 'enzyme';
import Highlighter from '../Highlighter';

describe('component::Highlighter', () => {
  it('Should match snapshoot', () => {
    const component = shallow(<Highlighter />);
    expect(component).toMatchSnapshot();
  });
});
