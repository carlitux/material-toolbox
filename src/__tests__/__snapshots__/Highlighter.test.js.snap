// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::Highlighter Should match snapshoot 1`] = `
<SyntaxHighlighter
  language="bash"
  showLineNumbers={true}
  style={
    Object {
      "hljs": Object {
        "background": "#23241f",
        "color": "#f8f8f2",
        "display": "block",
        "overflowX": "auto",
        "padding": "0.5em",
      },
      "hljs-addition": Object {
        "color": "#e6db74",
      },
      "hljs-attr": Object {
        "color": "#f92672",
      },
      "hljs-attribute": Object {
        "color": "#66d9ef",
      },
      "hljs-built_in": Object {
        "color": "#e6db74",
      },
      "hljs-builtin-name": Object {
        "color": "#e6db74",
      },
      "hljs-bullet": Object {
        "color": "#ae81ff",
      },
      "hljs-class .hljs-title": Object {
        "color": "#f8f8f2",
      },
      "hljs-code": Object {
        "color": "#a6e22e",
      },
      "hljs-comment": Object {
        "color": "#75715e",
      },
      "hljs-deletion": Object {
        "color": "#75715e",
      },
      "hljs-emphasis": Object {
        "color": "#a8a8a2",
        "fontStyle": "italic",
      },
      "hljs-keyword": Object {
        "color": "#f92672",
      },
      "hljs-link": Object {
        "color": "#ae81ff",
      },
      "hljs-literal": Object {
        "color": "#ae81ff",
      },
      "hljs-meta": Object {
        "color": "#75715e",
      },
      "hljs-name": Object {
        "color": "#f92672",
      },
      "hljs-number": Object {
        "color": "#ae81ff",
      },
      "hljs-params": Object {
        "color": "#f8f8f2",
      },
      "hljs-quote": Object {
        "color": "#ae81ff",
      },
      "hljs-regexp": Object {
        "color": "#ae81ff",
      },
      "hljs-section": Object {
        "color": "#a6e22e",
      },
      "hljs-selector-attr": Object {
        "color": "#e6db74",
      },
      "hljs-selector-class": Object {
        "color": "#a6e22e",
      },
      "hljs-selector-id": Object {
        "color": "#e6db74",
      },
      "hljs-selector-pseudo": Object {
        "color": "#e6db74",
      },
      "hljs-selector-tag": Object {
        "color": "#f92672",
      },
      "hljs-string": Object {
        "color": "#e6db74",
      },
      "hljs-strong": Object {
        "color": "#a8a8a2",
        "fontWeight": "bold",
      },
      "hljs-subst": Object {
        "color": "#f8f8f2",
      },
      "hljs-symbol": Object {
        "color": "#66d9ef",
      },
      "hljs-tag": Object {
        "color": "#f8f8f2",
      },
      "hljs-template-variable": Object {
        "color": "#e6db74",
      },
      "hljs-title": Object {
        "color": "#a6e22e",
      },
      "hljs-type": Object {
        "color": "#e6db74",
      },
      "hljs-variable": Object {
        "color": "#e6db74",
      },
    }
  }
/>
`;
